examples/arkode/CXX_parhyp/ark_heat2D_hypre_ls.cpp

```diff
@@ -382,6 +382,13 @@ int main(int argc, char* argv[])
   flag = SUNContext_Create(comm_w, &ctx);
   if (check_flag(&flag, "SUNContext_Create", 1)) { return 1; }
 
+  // Initialize hypre if v2.20.0 or newer
+#if HYPRE_RELEASE_NUMBER >= 22000 || SUN_HYPRE_VERSION_MAJOR > 2 || \
+  (SUN_HYPRE_VERSION_MAJOR == 2 && SUN_HYPRE_VERSION_MINOR >= 20)
```

Comment:
It would be great if we could have some of this mess inside the _hypre_ vector, so that our HYPRE-based examples could be simplified.  For example, if we could set a preprocessor directive `HYPRE_NEED_INIT`

Reply:
Unfortunately, this would complicate the auto-generated Makefiles. Ideally, we'll drop support for hypre < v2.21.0 at some point and this line can go away.